partcad/src/partcad/wrappers/custom_cqgi.py

```
@@ -0,0 +1,619 @@
+            env = (
+                EnvironmentBuilder()
+                .with_real_builtins()
+                .add_entry("__name__", "__cqgi__")
+                .add_entry("show_object", collector.show_object)
+                .add_entry("debug", collector.debug)
+                .add_entry("describe_parameter", collector.describe_parameter)
+                .build()
+            )
+
+            c = compile(self.ast_tree, CQSCRIPT, "exec")
+            exec(c, env)
```

Ruff (0.8.2)

Use of `exec` detected

(S102)